sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantCast.cs

```diff
@@ -106,19 +106,24 @@ private static void CheckExtensionMethodInvocation(SyntaxNodeAnalysisContext con
                 return;
             }
 
-            var elementType = GetElementType(invocation, methodSymbol, context.SemanticModel);
-            if (elementType == null)
+            if (!(methodSymbol.ReturnType is INamedTypeSymbol returnType) ||
+                !returnType.TypeArguments.Any())
```

Comment:
Extracting a method with this condition (`ReturnsGenericType`) would make the code easier to read.